### PR TITLE
Feature: Documentation

### PR DESCRIPTION
**Pull request by thomasdigby**
- Added documentation for Colours, Effects, Layout & space, and Typography
- Requires an update to the documentation site to handle `ColorPalette` and `Scale`
- There's lots of extra detail to fill in here and some pages that just have the theme information for now

Screenshot_2021-03-18 Screenshot(1)
Screenshot_2021-03-18 Screenshot(2)
Screenshot_2021-03-18 Screenshot(3)
Screenshot_2021-03-18 Screenshot(4)

### CODE REVIEW

File: src/documentation/LayoutAndSpace.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+</Scale>
+
+As an example, the following "Avatar" style component is created using the size and space scale. A `size` of `$3` is used to set the `height` and `width`, and `mr` and `mb` are used to set varying values from the space scale.
+
+```jsx preview
+<Flex css={{ alignItems: 'center' }}>
+  <Box
+    css={{
+      bg: '$primary900',
+      borderRadius: '50%',
```
Comment (andoulla) [inline src/documentation/LayoutAndSpace.mdx:43]: Just a thought, why not have the `50%` value in the radii options. It is used everywhere and it creates more consistency

File: src/documentation/Effects.mdx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+title: Effects
+---
```
Comment (andoulla) [inline src/documentation/Effects.mdx:3]: Should we have a small description here?

File: src/documentation/LayoutAndSpace.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    <Text size="sm" css={{ color: '$tonal700' }}>
+      @SteveBrusatte
+    </Text>
+  </Flex>
+</Flex>
+```
+
+## Radius
+
+The border-radii scale uses similar scale values for applying a border-radius.
```
Comment (hollg) [inline src/documentation/LayoutAndSpace.mdx:61]: Is it not just the `radii` scale?

File: src/documentation/Effects.mdx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+## Shadows
+
+<Scale scale={theme.shadows} css={{ mb: '$5' }}>
```
Comment (andoulla) [inline src/documentation/Effects.mdx:7]: Where is this `Scale` component coming from?

File: src/documentation/Typography.mdx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+title: Typography
+description: For consistent typography throughout the application we created a variety of Text components, you can read more about those specific font styles in the Text Component documentation.
```
Comment (hollg) [inline src/documentation/Typography.mdx:3]:
```suggestion
description: We have a variety of Text components to provide consistent typography throughout the application. You can read more about those specific font styles in the Text Component documentation.
```

File: src/documentation/Typography.mdx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        Abc-Xyz
+      </Text>
+      <Text size="lg">{value}</Text>
+    </Box>
+  )}
+</Scale>
+
+## Sizes
+
+This typographic sizing scale represents the `font-size` values used in our components, it loosely follows the [Perfect Fourth](https://type-scale.com/?scale=1.333&font=Inter) scale and can be accessed by using t-shirt notation
```
Comment (hollg) [inline src/documentation/Typography.mdx:28]:
```suggestion
This typographic sizing scale represents the `font-size` values used in our components. It loosely follows the [Perfect Fourth](https://type-scale.com/?scale=1.333&font=Inter) scale and can be accessed by using t-shirt notation
```

File: src/documentation/Typography.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+<Scale scale={theme.fontSizes} css={{ mb: '$5' }}>
+  {(value) => <Text css={{ fontSize: value }}>{value}</Text>}
+</Scale>
+
+An important thing to note is that most of our typography components, `Text`, `Heading`, `Label` and others, use [capsize](https://seek-oss.github.io/capsize/) to offset the white-space between the cap-height and the line-height. This means that spacing in and around our text will be tight to the characters.
```
Comment (hollg) [inline src/documentation/Typography.mdx:34]:
```suggestion
Note that most of our typography components (`Text`, `Heading`, `Label` and others) use [capsize](https://seek-oss.github.io/capsize/) to offset the white-space between the cap-height and the line-height. This means that spacing in and around our text will be tight to the characters.
```

**Review by hollg — CHANGES_REQUESTED** [review]

File: src/documentation/LayoutAndSpace.mdx (hunk 1 above)
Comment (thomasdigby) [reply src/documentation/LayoutAndSpace.mdx:43]: Yeah I think so, will add as a ticket

File: src/documentation/Effects.mdx (hunk 2 above)
Comment (thomasdigby) [reply src/documentation/Effects.mdx:3]: We're missing documentation on a few things, I'm keen to get this in and then add to it in a separate PR so we can get a first draft of the theme documentation live.

File: src/documentation/Effects.mdx (hunk 4 above)
Comment (thomasdigby) [reply src/documentation/Effects.mdx:7]: This is coming from the documentation site, PR for that incoming

**Review by hollg — APPROVED** [review]

**Pull request merged**